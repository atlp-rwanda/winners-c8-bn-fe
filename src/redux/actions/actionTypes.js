export const FECTH_REQUESTS = 'FECTH_REQUESTS';
export const FECTH_REQUESTS_COMMENTS = 'FECTH_REQUESTS_COMMENTS';
export const FECTH_REQUESTS_COMMENTS_FAILED = 'FECTH_REQUESTS_COMMENTS_FAILED';
export const POST_REQUESTS_COMMENT = 'FECTH_REQUESTS_COMMENTS_FAILED';
export const POST_REQUESTS_COMMENT_FAILED = 'FECTH_REQUESTS_COMMENTS_FAILED';
export const DELETE_REQUESTS_COMMENT = 'DELETE_REQUESTS_COMMENT';
export const DELETE_REQUESTS_COMMENT_FAILED = 'DELETE_REQUESTS_COMMENT_FAILED';

export const GOOGLE_SIGN_IN_START = 'GOOGLE_SIGN_IN_START';
export const GOOGLE_SIGN_IN_SUCCESS = 'GOOGLE_SIGN_IN_SUCCESS';
export const GOOGLE_SIGN_IN_FAIL = 'GOOGLE_SIGN_IN_FAIL';

export const FACEBOOK_SIGN_IN_START = 'FACEBOOK_SIGN_IN_START';
export const FACEBOOK_SIGN_IN_SUCCESS = 'FACEBOOK_SIGN_IN_SUCCESS';
export const FACEBOOK_SIGN_IN_FAIL = 'FACEBOOK_SIGN_IN_FAIL';

export const FETCH_ACCOMMODATIONS_LOADING = 'FETCH_ACCOMMODATIONS_LOADING';
export const FETCH_ACCOMMODATIONS_SUCCESS = 'FETCH_ACCOMMODATIONS_SUCCESS';
export const FETCH_ACCOMMODATIONS_FAILED = 'FETCH_ACCOMMODATIONS_FAILED';

export const FETCH_SINGLE_ACCOMMODATION_LOADING =
  'FETCH_SINGLE_ACCOMMODATION_LOADING';
export const FETCH_SINGLE_ACCOMMODATION_SUCCESS =
  'FETCH_SINGLE_ACCOMMODATION_SUCCESS';
export const FETCH_SINGLE_ACCOMMODATION_FAILED =
  'FETCH_SINGLE_ACCOMMODATION_FAILED';

export const FETCH_USER_LOADING = 'FETCH_USER_LOADING';
export const FETCH_USER_SUCCESS = 'FETCH_USER_SUCCESS';
export const FETCH_USER_FAILED = 'FETCH_USER_FAILED';

export const FETCH_ROLE_LOADING = 'FETCH_ROLE_LOADING';
export const FETCH_ROLE_SUCCESS = 'FECTH_ROLE_SUCCESS';
export const FETCH_ROLE_FAILED = 'FECTH_ROLE_FAILED';

export const ASSIGN_ROLE_LOADING = 'ASSIGN_ROLE_LOADING';
export const ASSIGN_ROLE_SUCCESS = 'ASSIGN_ROLE_SUCCESS';
export const ASSIGN_ROLE_FAILED = 'ASSIGN_ROLE_FAILED';

export const LOGOUT_SUCCESS = 'LOGOUT_SUCCESS';
export const LOGOUT_FAILED = 'LOGOUT_FAILED';

export const ACCOMMODATION_UPDATE_LOADING = 'ACCOMMODATION_UPDATE_LOADING';
export const ACCOMMODATION_UPDATE_SUCCESS = 'ACCOMMODATION_UPDATE_SUCCESS';
export const ACCOMMODATION_UPDATE_FAILURE = 'ACCOMMODATION_UPDATE_FAILURE';

export const ACCOMMODATION_DELETE_LOADING = 'ACCOMMODATION_DELETE_LOADING';
export const ACCOMMODATION_DELETE_SUCCESS = 'ACCOMMODATION_DELETE_SUCCESS';
export const ACCOMMODATION_DELETE_FAILURE = 'ACCOMMODATION_DELETE_FAILURE';

export const ACCOMMODATION_VIEW_LOADING = 'ACCOMMODATION_VIEW_LOADING';
export const ACCOMMODATION_VIEW_SUCCESS = 'ACCOMMODATION_VIEW_SUCCESS';
export const ACCOMMODATION_VIEW_FAILURE = 'ACCOMMODATION_VIEW_FAILURE';
export const APPROVEREJECT_PENDING_REQUEST = 'APPROVE_PENDING_REQUEST';
export const APPROVEREJECT_SUCCESS = 'APPROVE_REQUEST_SUCCESS';
export const APPROVEREJECT_ERROR = 'APPROVE_REQUEST_ERROR';

export const FETCH_ALL_CHATS = 'GET_CHATS';
export const FETCH_ALL_CHATS_FAILED = 'FAILED_TO_LOAD_CHATS';
export const CHAT_SUCCESS = 'CHAT_SUCCESS';
export const CHATTING_STATUS = 'CHATTING_STATUS';
