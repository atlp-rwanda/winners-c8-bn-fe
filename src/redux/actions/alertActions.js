
export const successAlert = (message) => {
    return {
        type: 'success-alert',
        payload: message
    }
    
};


export const failureAlert = (message) => {
    return {
        type: 'failure-alert',
        payload: message
    }
};
