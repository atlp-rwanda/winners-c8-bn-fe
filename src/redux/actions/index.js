// this folder will contain redux's actions