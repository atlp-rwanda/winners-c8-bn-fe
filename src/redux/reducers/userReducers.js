export function registration(state={}, action){
    switch (action.type) {
        case 'REGISTER_USER':
        return { 
            ...state,
            registering: true,
        };

        case 'FAILURE':
        return { 
            ...state,
            registering: false,

        };

        case 'USER_EXIST':
        return {
            ...state,
            registering: false
        };

        case 'SUCCESS':
        return { 
            ...state,
            registering: false,
        };

        default:
        return state
    }

}
