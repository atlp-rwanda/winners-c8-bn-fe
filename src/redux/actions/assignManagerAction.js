import { successToast, errorToast } from '../../helpers/generateToast';
import axiosInstance from '../../helpers/http';
import { FETCH_MANAGERS ,FETCH_MANAGERS_FAILED, ASSIGN_MANAGER ,ASSIGN_MANAGER_FAILED } from './actionTypes'


  export const fetchManagers = () => async (dispatch) => {
    return await 
    axiosInstance.get('/users/managers')
      .then((res) => {
        dispatch({
          type: FETCH_MANAGERS,
          payload: res.data,
        });
      })
      .catch((err) => {
        dispatch({
          type: FETCH_MANAGERS_FAILED,
          payload: 'Fetch Managers failed',
        });
      });
      
  };


  export const assignManager = (SelectedEmail, SelectedManager) => async (dispatch) => {
    return await 
    axiosInstance.patch('/users/assignManager',{email: SelectedEmail, managerId: SelectedManager})
      .then((res) => {
        if(res.status == 200){
          dispatch({
            type: ASSIGN_MANAGER,
            payload: res.data,
          });
          successToast(res.data.message)
          dispatch(fetchManagers());

        }else{
          errorToast("Only Admin can do this")
        }
      })
      .catch((err) => {
        dispatch({
          type: ASSIGN_MANAGER_FAILED,
          payload: 'Assigning Manager failed',
        });
      });
      
  };