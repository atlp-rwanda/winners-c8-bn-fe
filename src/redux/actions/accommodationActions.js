import axios from 'axios'
import { successToast, errorToast } from '../../helpers/generateToast';
import { successAlert, failureAlert} from './alertActions';
import { FETCH_ACCOMMODATIONS_LOADING, 
         FETCH_ACCOMMODATIONS_SUCCESS, 
         FETCH_ACCOMMODATIONS_FAILED,
         FETCH_SINGLE_ACCOMMODATION_LOADING,
         FETCH_SINGLE_ACCOMMODATION_SUCCESS,
         FETCH_SINGLE_ACCOMMODATION_FAILED} from "./actionTypes";
import {accommodationsUrl} from "../utils/apiUrls";
import {authHeader} from '../utils/dataSession';

export const listAccommodations = () => async dispatch => {
    dispatch({
        type: FETCH_ACCOMMODATIONS_LOADING
    });

  return await axios.get(accommodationsUrl, { headers: authHeader() })
      .then(res => {
          dispatch({
              type: FETCH_ACCOMMODATIONS_SUCCESS,
              payload: res.data
          });
      })
      .catch(err => {
          dispatch({
              type: FETCH_ACCOMMODATIONS_FAILED,
              payload: err.message,
              message: err
          });
          errorToast(err)
      })
}

export const detailsAccommodation = (accommodationId) => async dispatch => {
    dispatch({
        type: FETCH_SINGLE_ACCOMMODATION_LOADING
    });

  return await axios.get(accommodationsUrl+`${accommodationId}`, { headers: authHeader() })
      .then(res => {
          dispatch({
              type: FETCH_SINGLE_ACCOMMODATION_SUCCESS,
              payload: res.data
          });
      })
      .catch(err => {
          dispatch({
              type: FETCH_SINGLE_ACCOMMODATION_FAILED,
              payload: "failed to fetch",
              message: err
          });
      })
}
