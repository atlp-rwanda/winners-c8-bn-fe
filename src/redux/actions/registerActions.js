import axios from 'axios';
import { successToast, errorToast } from '../../helpers/generateToast';
import { successAlert } from './alertActions';

const apiUrl = 'https://winners-c8-bn-be-staging.herokuapp.com/api';

export const register = ({ firstName, lastName, email, password }) => {
  let res;
  return (dispatch) => {
    dispatch({
      type: 'REGISTER_USER',
    });

    axios({
      method: 'POST',
      url: `${apiUrl}/auth/register`,
      data: { firstName, lastName, email, password },
      validateStatus: () => true,
    })
      .then((res) => {
        if (res.data.error) {
          dispatch({
            type: 'FAILURE',
            payload: res.data.error,
          });
          errorToast(res.data.error);
        }
        if (res.data.message === 'Ooops! User already exists!') {
          dispatch({
            type: 'USER_EXIST',
            payload: res.message,
          });
          errorToast(res.data.message);
        }
        if (res.data.success && res.data.status === 201) {
          dispatch({
            type: 'SUCCESS',
            payload: res.data,
          });
          dispatch(successAlert(res.data.message));
          // successToast(res.data.message)
          history.push({ pathname: '/login', state: 'verify' });
        }
      })
      .catch((error) => {
        dispatch({
          type: 'FAILURE',
          payload: error.toString(),
        });
        errorToast(error.message);
      });
  };
};

export const userActions = {
  register,
};
