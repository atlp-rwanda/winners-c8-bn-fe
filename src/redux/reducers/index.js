// this folder will contain redux's reducers
import { combineReducers } from 'redux';
import authReducer from './authReducer';
import navbarReducer from './navbarReducer';
import { registration } from './userReducers';
import { alert } from './alertReducers';
import requestReducer from './requestReducer';
import resetReducer from './resetReducer';
import recoverReducer from './recoverReducer';
import userReducer from './SocialReducers';
import {
  fetchUserProfileReducer,
  updateUserProfileReducer,
} from './userProfileReducer';

export default combineReducers({
  auth: authReducer,
  navbar: navbarReducer,
  requests: requestReducer,
  userProfile: fetchUserProfileReducer,
  updateUserProfile: updateUserProfileReducer,
  registration,
  alert,
  reset: resetReducer,
  recover: recoverReducer,
  user: userReducer,
});
