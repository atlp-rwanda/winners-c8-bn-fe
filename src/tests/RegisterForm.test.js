/**
 * @jest-environment jsdom
 */
import React from "react";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import { register } from "../../src/redux/actions/registerActions";
const middleware = [thunk];
const mockStore = configureStore(middleware);
const mock = new MockAdapter(axios);
const store = mockStore({});
import { registration } from "../../src/redux/reducers/userReducers";
import { cleanup } from "@testing-library/react";
import UserRegisterForm from "../components/RegisterForm";
import { Provider } from "react-redux";
import { BrowserRouter as Router, MemoryRouter as MemoryRouter} from "react-router-dom";
import "@testing-library/jest-dom";
import renderer from 'react-test-renderer';

const apiUrl ='https://winners-c8-bn-be-staging.herokuapp.com/api/auth/register';

afterEach(()=>{
	cleanup()
})

describe('Connected Register Form React-Redux Component', () => {
	let store;
	let component;
  
	beforeEach(() => {
	  store = mockStore({
		registration: false,
		alert: ''
	  });

	  store.dispatch = jest.fn();

	  component = renderer.create(
		<Provider store={store}>
			<MemoryRouter>
				<UserRegisterForm />
			</MemoryRouter>
		  
		</Provider>
	  );
	});
  
	it('should render with given state from Redux store', () => {
		expect(component.toJSON()).toMatchSnapshot();
	});
  });



describe('register(user) for actions', () => {

	it('dispatches SUCCESS after signup', async () => {
		store.clearActions();
		const userObj = {
            firstName:'kamali', 
            lastName:'elia',
            email:`kamali${new Date().getMilliseconds()}@gmail.com`,
			password: 'Password@12345',
		};

		await store.dispatch(register(userObj));

	})
	

    it('dispatches FAILURE after signup failure', async () => {
		store.clearActions();
		const userObj = {
            firstName:'kamali', 
            lastName:'elia',
            email:`kamali${new Date().getMilliseconds()}@gmail.com`,
			password: 'password@12345',
		};

		await store.dispatch(register(userObj));

	})
})


describe('registration(state, action)', ()=>{
	const state = {}

	const RegisterUserAction = {
		type: 'REGISTER_USER'
	}

	const FailureAction = {
		type: 'FAILURE',
		payload: 'payload'
	}

	const SuccessAction = {
		type: 'SUCCESS',
		payload: 'payload'
	}

	const userExistAction = {
		type: 'USER_EXIST',
        payload: 'payload'
	}

	it('should return registering true',()=>{
        const updatedState = registration(state, RegisterUserAction);
        expect(updatedState).toEqual({registering: true});
    })

	it('should return registering false on failure',()=>{
        const updatedState = registration(state, FailureAction);
        expect(updatedState).toEqual({registering: false});
    })

	it('should return registering false on success',()=>{
        const updatedState = registration(state, SuccessAction);
        expect(updatedState).toEqual({registering: false});
    })

	it('should return registering false on user exists',()=>{
        const updatedState = registration(state, userExistAction);
        expect(updatedState).toEqual({registering: false});
    })

})



