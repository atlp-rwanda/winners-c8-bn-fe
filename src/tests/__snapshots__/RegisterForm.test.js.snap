// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected Register Form React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="register-section pt-100 pb-100 md-pt-80 md-pb-80"
>
  <div>
    <div
      className="Toastify"
    />
  </div>
  <div
    className="registration"
  >
    <div
      className="register-box"
    >
      <div
        className="sec-title text-center mb-30"
      >
        <h5
          className="title 200 mb-10"
        >
          Create New Account.
        </h5>
      </div>
      <div
        className="styled-form"
      >
        <div
          id="form-messages"
        />
        <form
          id="register-form"
          method="post"
          onSubmit={[Function]}
        >
          <div
            className="row clearfix"
          >
            <div
              className="form-group col-lg-12 mb-25"
            >
              <input
                className="field"
                id="Name"
                name="firstName"
                onChange={[Function]}
                placeholder="First Name"
                type="text"
                value=""
              />
            </div>
            <div
              className="form-group col-lg-12"
            >
              <input
                className="field"
                id="last"
                name="lastName"
                onChange={[Function]}
                placeholder="Last Name"
                type="text"
                value=""
              />
            </div>
            <div
              className="form-group col-lg-12"
            >
              <input
                className="field"
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Email address"
                type="email"
                value=""
              />
            </div>
            <div
              className="form-group col-lg-12"
            >
              <input
                className="field"
                id="puser"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />
            </div>
            <div
              className="form-group col-lg-12"
            >
              <input
                className="field"
                id="Confirm"
                name="confPassword"
                onChange={[Function]}
                placeholder="Confirm Password"
                type="password"
                value=""
              />
            </div>
            <div
              className="form-group col-lg-12 col-md-12 col-sm-12 text-center"
            >
              <div>
                <input
                  className="btn"
                  type="submit"
                  value="Sign Up"
                />
              </div>
            </div>
            <div
              className="row d-flex justify-content-center"
            >
              <div
                className="col-xs-12"
              >
                <h5
                  className="header5"
                >
                  _________
                </h5>
              </div>
              <div
                className="col-xs-12 col-md-8"
              >
                <h5
                  className="header5"
                >
                  Or Sign Up with:
                </h5>
              </div>
            </div>
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginTop": "10px",
                }
              }
            >
              <div
                style={
                  Object {
                    "paddingRight": "20px",
                  }
                }
              >
                <a
                  href="undefined/oauth/google"
                >
                  <svg
                    enableBackground="new 0 0 48 48"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                        "cursor": "pointer",
                        "fontSize": "40px",
                      }
                    }
                    version="1.1"
                    viewBox="0 0 48 48"
                    width="1em"
                    x="0px"
                    xmlns="http://www.w3.org/2000/svg"
                    y="0px"
                  >
                    <path
                      d="M43.611,20.083H42V20H24v8h11.303c-1.649,4.657-6.08,8-11.303,8c-6.627,0-12-5.373-12-12
	c0-6.627,5.373-12,12-12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657C34.046,6.053,29.268,4,24,4C12.955,4,4,12.955,4,24
	c0,11.045,8.955,20,20,20c11.045,0,20-8.955,20-20C44,22.659,43.862,21.35,43.611,20.083z"
                      fill="#FFC107"
                    />
                    <path
                      d="M6.306,14.691l6.571,4.819C14.655,15.108,18.961,12,24,12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657
	C34.046,6.053,29.268,4,24,4C16.318,4,9.656,8.337,6.306,14.691z"
                      fill="#FF3D00"
                    />
                    <path
                      d="M24,44c5.166,0,9.86-1.977,13.409-5.192l-6.19-5.238C29.211,35.091,26.715,36,24,36
	c-5.202,0-9.619-3.317-11.283-7.946l-6.522,5.025C9.505,39.556,16.227,44,24,44z"
                      fill="#4CAF50"
                    />
                    <path
                      d="M43.611,20.083H42V20H24v8h11.303c-0.792,2.237-2.231,4.166-4.087,5.571
	c0.001-0.001,0.002-0.001,0.003-0.002l6.19,5.238C36.971,39.205,44,34,44,24C44,22.659,43.862,21.35,43.611,20.083z"
                      fill="#1976D2"
                    />
                  </svg>
                </a>
              </div>
              <div
                style={
                  Object {
                    "paddingLeft": "20px",
                  }
                }
              >
                <a
                  href="undefined/oauth/facebook"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                        "cursor": "pointer",
                        "fontSize": "35px",
                      }
                    }
                    version="1.1"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M9.5 3h2.5v-3h-2.5c-1.93 0-3.5 1.57-3.5 3.5v1.5h-2v3h2v8h3v-8h2.5l0.5-3h-3v-1.5c0-0.271 0.229-0.5 0.5-0.5z"
                    />
                  </svg>
                </a>
              </div>
            </div>
            <div
              className="form-group col-lg-12 col-md-12 col-sm-12"
            >
              <div
                className="notify"
              >
                Already have an account? 
                <a
                  href="/login"
                  onClick={[Function]}
                >
                  Sign In
                </a>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
